DDDEastAnglia/Services/Messenger/Email/MailMessage.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Net.Mail;
+
+namespace DDDEastAnglia.Services.Messenger.Email
+{
+    public class MailMessage
+    {
+        protected bool Equals(MailMessage other)
+        {
+            return Equals(From, other.From) && Equals(To, other.To) && string.Equals(Subject, other.Subject) && string.Equals(Body, other.Body);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != this.GetType()) return false;
+            return Equals((MailMessage)obj);
+        }
+
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                int hashCode = (From != null ? From.GetHashCode() : 0);
+                hashCode = (hashCode * 397) ^ (To != null ? To.GetHashCode() : 0);
+                hashCode = (hashCode * 397) ^ (Subject != null ? Subject.GetHashCode() : 0);
+                hashCode = (hashCode * 397) ^ (Body != null ? Body.GetHashCode() : 0);
+                return hashCode;
+            }
+        }
```

Comment:
No, see the commit history. In order for NSubstitute to be able to assert that a particular `MailMessage` was delivered, we need to be able to compare two instances of `MailMessage` for equality. It seems that NSubstitute doesn't support alternative methods of equality comparison, so we have to go with this.

It's not that nice, I agree, but I think we're stuck with it.